lib/JSON/Validator/OpenAPI/Mojolicious.pm

```diff
@@ -19,6 +19,28 @@ has version => 2;
 
 sub E { JSON::Validator::Error->new(@_) }
 
+has generate_definitions_path => sub {
+  my $self = shift;
+  Scalar::Util::weaken($self);
+
+  return sub {
+    my $ref = shift;
+    my @path;
+
+    # Try to determin the path from the fqn
+    if ($ref->fqn =~ m|^(.*)#/(.+)$|) {
+      my $relative = $1;
+      my $path     = $2;
+
+      @path = split '/', $path;
+      pop @path;
```

Comment:
Can you do this instead to make the code tidier?

```
my ($relative, @path) = ($1, split '/', $2);
```

Reply:
fixed in upcoming push

Reply:
You still need to pop the last part of the path, but what about this instead ?

`my $path = Mojo::Path->new( $ref->fqn =~ m!^.*#/(.+)$! )->to_dir->parts;`

And the whole sub then becomes :

```perl
has generate_definitions_path => sub {
  my $self = shift;
  Scalar::Util::weaken($self);

  return sub {
    my $ref = shift;

    # Try to determine the path from the fqn
    # We are only interested in the path in the fqn, so following fqn:
    #
    # #/components/schemas/some_schema, the returned path with be ['components', 'schemas']   (v3)
    # #/definitions/some_schema, the returned path with be ['definitions']  (v2)

    my $path = Mojo::Path->new( $ref->fqn =~ m!^.*#/(.+)$! )->to_dir->parts;
    return $path->[0] ? $path : ['definitions'];
  };
};
```

Reply:
I get the wrong button. Not very good UX on my screen I think :/

Anyhow: I think maybe the Mojo::Path solution is slower, but I’m not going to benchmark it, and it’s probably irrelevant, so I’m fine with your proposal.

Reply:
I get the wrong button. Not very good UX on my screen I think :/

Anyhow: I think maybe the Mojo::Path solution is slower, but I’m not going to benchmark it, and it’s probably irrelevant, so I’m fine with your proposal.

Reply:
If this turns out to be slowing things down, not my impression though. Its an easy fix ;)